interfaces/builtin/multipass_support.go

```diff
@@ -0,0 +1,129 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2019 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package builtin
+
+/*
+ * Multipass is a tool to create and manage Virtual Machines and their images.
+ * Each VM runs as a separate "qemu" process (on Linux). VM images are automatically
+ * downloaded, but need conversion using "qemu-img". Networking is provided by
+ * configuring a TUN/TAP network on the host, with DHCP provided by a shared
+ * "dnsmasq" process. File sharing between the VM and the host is provided by a
+ * "sshfs_server" utility. All these utilities are shipped in the snap.
+ *
+ * Each of these utilities have a very different different purpose, and an attempt
+ * to confine them all in a single profile would result in an extremely broad AppArmor
+ * profile.
+ *
+ * Instead we defer to Multipass the responsibility of generating custom AppArmor
+ * profiles for each of these utilities, and trust it launches each utility with
+ * all possible security mechanisms enabled. The Multipass daemon itself will run
+ * under this restricted policy.
```

Comment:
As a meta-comment. I would love to explore a framework where snapd would be aware of this and could be in the loop. Specifically where snapd would receive the profile from multipass, associate it with a security tag (also sent by multipass) and then generate the profiles. I don't mean to block this by any chance but if we grow more systems like that (perhaps LXD already qualifies) I think it could be beneficial to the overall integrity of the sandbox mechanism.

Reply:
Yes agreed, I would like to see something like this.

Reply:
Maybe it's too strong coupling, but multipass could work with something like this: the snapd multipass interface contains AA & seccomp profile templates, which have placeholders for strings/file paths. snapd would be able to check those strings/file paths for safety. 

Multipass could then ask snapd to run a process it specifies, under those AA/seccomp profiles with placeholders it requests.

Reply:
Whenever you are at the point of thinking through this, please include me. I think I'd rather see snapd providing the template for the VMs and then multipassd asking snapd to load one with a given profile name. Then multipassd doesn't need mac_admin at all and it isn't supplying any policy itself.